src/giftCards/GiftCardUpdatePage/messages.ts

```diff
@@ -1,12 +1,34 @@
-import { defineMessages } from "react-intl";
+import { commonMessages } from "@saleor/intl";
+import { GiftCardErrorCode } from "@saleor/types/globalTypes";
+import commonErrorMessages from "@saleor/utils/errors/common";
+import { defineMessages, IntlShape } from "react-intl";
+
+import { GiftCardUpdate_giftCardUpdate_errors } from "./types/GiftCardUpdate";
 
 export const giftCardUpdateDetailsCardMessages = defineMessages({
   title: {
     defaultMessage: "Details",
     description: "GiftCardUpdateDetailsCard title"
-  },
-  setBalanceButtonLabel: {
-    defaultMessage: "Set balance",
-    description: "GiftCardUpdateDetailsCard set balance button label"
   }
 });
+
+export function getGiftCardErrorMessage(
+  error: Omit<GiftCardUpdate_giftCardUpdate_errors, "__typename"> | undefined,
+  intl: IntlShape
+): string {
+  if (error) {
+    switch (error.code) {
+      case GiftCardErrorCode.GRAPHQL_ERROR:
```

Comment:
This doesn't seem like a good error for input level field 🤔

Reply:
We use this logic in multiple places, maybe we should remove it from all of them then? 🤔

Reply:
Create a ticket for this in 3.1/maintenance list and leave it as is right now